shared/Microsoft.Extensions.CopyOnWriteDictionary.Sources/CopyOnWriteDictionary.cs

```diff
@@ -13,6 +13,10 @@ internal class CopyOnWriteDictionary<TKey, TValue> : IDictionary<TKey, TValue>
         private readonly IEqualityComparer<TKey> _comparer;
         private IDictionary<TKey, TValue> _innerDictionary;
 
+        public CopyOnWriteDictionary(IDictionary<TKey, TValue> sourceDictionary)
```

Comment:
@NTaylorMullen brings up a good point. This overload will provide a way to shoot oneself in the foot when the source dictionary's comparer is not the default one i.e. `new CopyOnWriteDictionary<string, object>(new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase))`. In those cases, the second overload should be used.  Assuming we are the only consumers of this internal class and we don't use it incorrectly (famous last words), it's not too big of a problem.

Another proposal was to add an overload of `CopyOnWriteDictionary(Dictionary<TK, TV> source) : this(source, source.Comparer)` but this is rather limited since we usually pass in an `IDictionary` interface instead of the concrete type. Also, we may be wrapping COWDictionary multiple times and it only implements `IDictionary<TK, TV>`. 

All in all, this may be more trouble than its worth.